### PR TITLE
Show_pw

### PR DESCRIPTION
**Pull request by bstutes**
Show password input (checkbox) #1984

This pull request updates/supersedes my pull request of June 9, 2017.

Details of changes follow.

NOTE: All "-js-pw-m-*" CSS selectors referenced are
  for use by the password reveal JS event handler
  code.

Clean-up to pass "make lint" and remove a couple of
files that should not have been included in the
original pull request.

.../tests/unit:
 test_csp.py:
  Updated script-src assert statement spec to include "ajax.googleapis.com".

.../warehouse/templates/account:
 login.html, register.html:
  Added "-js-pw-m-fld" CSS selector to form.password
  class_ string for PW reveal/hide toggle.
  Added label and checkbox elements for triggering
  password reveal function. Checkbox element 
  identified by "-js-pw-m-cbx" CSS selector for
  cleanup form on submit.
  Added "-js-pw-m-sbt" selector to form submit
  button used by JS event handler to trigger
  form cleanup on submit.
  Added "extra_js" block to insert "jQuery" and
  "pwmask" script tags for password reveal JS code.
  In register.html, "for=" attribute for the email input
  was incorrect. Changed from "full_name" to "email".

.../warehouse/static/js:
 pwmask/pw-mask-toggle.js
  JS containing event handlers for password
  reveal/hide toggle and submit cleanup.

.../warehouse:
 csp.py:
  Added "ajax.googleapis.com" script-src policy
  statement to allow loading jquery.min.js in
  login.html and register.html.

.../Gulpfile.babel.js:
  Added path.join statement for "pw-mask-toggle.js"
  to files list in gulp.task("dist:js", ...) function.
  Added "else if (relPath..." statement for naming
  minified JS code for password reveal toggling.

### CODE REVIEW

**Comment by nlhkabu** [comment]
@bstutes - can you please post some screenshots of the affected pages?

**Comment by nlhkabu** [comment]
oh, and thankyou for this PR! :tada:

**Comment by bstutes** [comment]
@nlhkabu - Apologies for the delay in responding.  Here are the screenshots you requested.
pypi-login
pypi-register

**Comment by nlhkabu** [comment]
Thanks @bstutes and sorry for the delay in my response.

I think it would be better to group the checkbox with the label of field.  This way it is more clearly grouped with the password:

screenshot from 2017-07-15 15-52-11

If you'd prefer me to handle the CSS for this, we can merge this as is and I can make a separate PR :)

**Comment by bstutes** [comment]
Hello @nlhkabu.  Since I am not familiar with using SASS it would be better if you made the CSS changes required to move the password reveal checkbox to place as you noted above.

**Comment by nlhkabu** [comment]
ok on my side @dstufft - if you can please review the gulp file in particular, that would be great :)

**Comment by bstutes** [comment]
@nlhkabu I have been studying up on CSS and taken a brief glance and using SASS since we last communicated.  As a result, I have made further changes in this item.

1) Added "&__show-pw" selector to "static/sass/blocks/_form-group.scss" for the "Show Password" Checkbox.
2) Modified the code in "warehouse/templates/accounts/{login,register}.html" by inserting a <span> element that places the checkbox and the "Show password" item inside the <label> for the password fields.  The result is the pages looks as you suggested earlier.  Please see the attached screen shots.

Since I have not yet seen a commit for this pull request and I have since executed a fresh fork to catch up with the changes made to the pypa/warehouse master, then reimplemented all my changes under a branch of this new repo, with your permission I would like to close this pull request and open a new one that contains all the prior changes as well as my newest changes.

Please advise.

pypi_login_screen
pypi_register_screen

**Comment by bstutes** [comment]
It appears It was necessary to close this pull because I deleted my original fork of the application.
As such, I will push my branch with the changes to my repo and open a new pull request.

My apologies.
